cmd/client.go

```diff
@@ -0,0 +1,188 @@
+//		- probe.Init() error: an error occurred when the probe, created from the config passsed, was initialised.
+//			-> There was a problem in the Init() function of the probe type passed.
+//		- options.BuildProbeOptions() error: an error occurred when building the options for this probe from the config supplied.
+//			-> The options could not be built from the config supplied.
+func (client *CloudproberClient) addProbeFromConfig(ctx context.Context, probePb *configpb.ProbeDef) error {
+	// No mutex locking handled here as this is a private method called by
+	// a public method which handles locking.
+	_, err := client.client.AddProbe(ctx, &proberpb.AddProbeRequest{ProbeConfig: probePb})
+	return err
+}
+
+// RegisterAndAddProbe registers a probe type and adds the probe to Cloudprober to be run.
+// Parameters:
+// - probePb: - This probe config must be unmarshalled before being passed as an argument.
+// Returns:
+// - error:
+//	      - See addProbeFromConfig errors.
+func (client *CloudproberClient) RegisterAndAddProbe(extensionNumber int, ctx context.Context, probePb *configpb.ProbeDef, hermesProbeToAdd cpprobes.Probe) error {
+	client.clientMux.Lock()
+	defer client.clientMux.Unlock()
+
+	// First, register the probe as an extension with Cloudprober.
+	cpprobes.RegisterProbeType(extensionNumber, func() cpprobes.Probe { return hermesProbeToAdd })
+
+	// Add the probe to Cloudprober
+	// The probe will be scheduled and run by Cloudprober
+	// Adding a probe will consume the probe type registration.
+	// If you want to add multiple probes, you must register and add each one individually.
+	// Only one extension type can be registered at any given time.
+	// If more are registered, Cloudprober will throw an error.
+	return client.addProbeFromConfig(ctx, probePb)
+}
+
+// RemoveProbe removes a probe from Cloudprober, given the probe name (located in the probe config).
+// An error will be returned if there is no active probe with this probe name.
+// Parameters:
+// - probeName: This must be the name of an active probe in Cloudprober.
+// Returns:
+// - error:
+//	   - Code: 3, InvalidArgument: probeName is an empty string
+//	   - Code: 5,  NotFound: cannot find a probe matching this probe name
+func (client *CloudproberClient) RemoveProbe(probeName string) error {
+	client.clientMux.Lock()
+	defer client.clientMux.Unlock()
+
+	_, err := client.client.RemoveProbe(context.Background(), &proberpb.RemoveProbeRequest{ProbeName: &probeName})
+	return err
+}
+
+// ListProbes() returns a list of active probes from Cloudprober.
+// Returns:
+// - ListProbesResponse: This is the list of active probes in Cloudprober.
+//			 If this is empty, there are no active probes.
+// - Error:
+//	   - See Cloudprober ListProbes() RPC for details on an error.
+func (client *CloudproberClient) ListProbes() (*proberpb.ListProbesResponse, error) {
+	// client.clientMux.Lock()
```

Comment:
Delete these, rather than commenting them out.

Reply:
Deleted in latest commit.